### PR TITLE
Fix shenandoah conditional so it is correctly enabled

### LINKED ISSUE #2176: Shenandoah GC not available in 11.0.9
**Platform: Linux**

**Architecture: x86_64**

Please describe the issue that you are having with our binaries and if possible how to recreate it.

The AdoptOpenJDK 11.0.9 build doesn't appear to include Shenandoah GC.
Is it supposed to include the GC?

According to OpenJDK [wiki](https://wiki.openjdk.java.net/display/shenandoah/Main#Main-JDKSupport) it should be in mainline builds:
> _In mainline OpenJDK 11u builds since 11.0.9._

Here's what happens when trying to enable the GC on v11.0.9:
```
~ % docker run -it openjdk:11-jre-buster

root@6e987dc384f8:/# java -version
openjdk version "11.0.9" 2020-10-20
OpenJDK Runtime Environment 18.9 (build 11.0.9+11)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.9+11, mixed mode)

root@6e987dc384f8:/# java -XX:+UseShenandoahGC -Xlog:gc -version
Unrecognized VM option 'UseShenandoahGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

### PR DESCRIPTION
**Pull request by sxa**
Shenandoah was not being correctly enabled in https://github.com/AdoptOpenJDK/openjdk-build/pull/2125 due the the use of a numeric `-eq` comparison on a string field that was `jdk11`.

Error in the log:
`/home/adoptopenjdk/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-x64-hotspot/sbin/build.sh: line 81: [: jdk11: integer expression expected`

This fixes it so that Shendandoah will be correctly enabled on the builds where it should have been.

Fixes #2176 

Signed-off-by: Stewart X Addison <sxa@redhat.com>

### CODE REVIEW

**Comment by sxa** [comment]
run tests

**Review by smlambert — APPROVED** [review]

**Review by aahlenst — APPROVED** [review]

**Comment by sxa** [comment]
Checks are failing as a result of what seems an u related OpenJ9/aarch64 failure which is causing a segfault so should not block this PR

**Comment by sxa** [comment]
Merging this so I can attempt to kick off a new build

**Comment by marko-asplund** [comment]
@sxa Thanks for fixing the issue! 👍
Any approximate estimate for when a fixed build will be available?

**Comment by sxa** [comment]
> 
> 
> @sxa Thanks for fixing the issue! 👍
> Any approximate estimate for when a fixed build will be available?

Updated builds are complete and undergoing testing (Not that there should be any issue) - most of the way through. They'll show up first under the "build artifacts" in our CI when this job completes https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/1233/ then I need a second TSC approval to make them live so release time depends on who's awake :-)

**Comment by marko-asplund** [comment]
Thanks! Just tried out the build quickly and I'm at least able to start the JVM with Shenandoah GC.
My app's running in the `openjdk:11-jre-buster` docker container, so in order to test it properly I'd either need to build a custom container or wait for the  new AdoptOpenJDK build get included in the openjdk container.

**Pull request merged**
